DOC: doc build failing

Currently, the doc build is failing eg. https://app.circleci.com/pipelines/github/scipy/scipy/23282/workflows/5ec2135b-3be9-4fe0-9e1b-f800067ba42f/jobs/78346
Potentially an upstream issue with the pydata syphinx theme?

Let me copy the traceback for future reference:

```
Running Sphinx v7.2.6

...


Traceback (most recent call last):
  File "/home/circleci/.pyenv/versions/3.9.18/lib/python3.9/site-packages/sphinx/events.py", line 97, in emit
    results.append(listener.handler(self.app, *args))
  File "/home/circleci/.pyenv/versions/3.9.18/lib/python3.9/site-packages/pydata_sphinx_theme/__init__.py", line 145, in update_templates
    if theme_css_name in context["css_files"]:
  File "/home/circleci/.pyenv/versions/3.9.18/lib/python3.9/site-packages/sphinx/builders/html/_assets.py", line 40, in __eq__
    warnings.warn('The str interface for _CascadingStyleSheet objects is deprecated. '
sphinx.deprecation.RemovedInSphinx90Warning: The str interface for _CascadingStyleSheet objects is deprecated. Use css.filename instead.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/circleci/.pyenv/versions/3.9.18/lib/python3.9/site-packages/sphinx/cmd/build.py", line 298, in build_main
    app.build(args.force_all, args.filenames)
  File "/home/circleci/.pyenv/versions/3.9.18/lib/python3.9/site-packages/sphinx/application.py", line 355, in build
    self.builder.build_update()
  File "/home/circleci/.pyenv/versions/3.9.18/lib/python3.9/site-packages/sphinx/builders/__init__.py", line 293, in build_update
    self.build(to_build,
  File "/home/circleci/.pyenv/versions/3.9.18/lib/python3.9/site-packages/sphinx/builders/__init__.py", line 363, in build
    self.write(docnames, list(updated_docnames), method)
  File "/home/circleci/.pyenv/versions/3.9.18/lib/python3.9/site-packages/sphinx/builders/__init__.py", line 568, in write
    self._write_parallel(sorted(docnames),
  File "/home/circleci/.pyenv/versions/3.9.18/lib/python3.9/site-packages/sphinx/builders/__init__.py", line 595, in _write_parallel
    self.write_doc(firstname, doctree)
  File "/home/circleci/.pyenv/versions/3.9.18/lib/python3.9/site-packages/sphinx/builders/html/__init__.py", line 655, in write_doc
    self.handle_page(docname, ctx, event_arg=doctree)
  File "/home/circleci/.pyenv/versions/3.9.18/lib/python3.9/site-packages/sphinx/builders/html/__init__.py", line 1107, in handle_page
    newtmpl = self.app.emit_firstresult('html-page-context', pagename,
  File "/home/circleci/.pyenv/versions/3.9.18/lib/python3.9/site-packages/sphinx/application.py", line 492, in emit_firstresult
    return self.events.emit_firstresult(event, *args,
  File "/home/circleci/.pyenv/versions/3.9.18/lib/python3.9/site-packages/sphinx/events.py", line 118, in emit_firstresult
    for result in self.emit(name, *args, allowed_exceptions=allowed_exceptions):
  File "/home/circleci/.pyenv/versions/3.9.18/lib/python3.9/site-packages/sphinx/events.py", line 108, in emit
    raise ExtensionError(__("Handler %r for event %r threw an exception") %
sphinx.errors.ExtensionError: Handler <function update_templates at 0x7f080d5f5f70> for event 'html-page-context' threw an exception (exception: The str interface for _CascadingStyleSheet objects is deprecated. Use css.filename instead.)

Extension error (pydata_sphinx_theme):
Handler <function update_templates at 0x7f080d5f5f70> for event 'html-page-context' threw an exception (exception: The str interface for _CascadingStyleSheet objects is deprecated. Use css.filename instead.)
make: *** [Makefile:114: html-build] Error 2
```

We're pinning to an old `pydata-sphinx-theme` (0.9.0) and `sphinx` 7.2.6 has been released for 2 months. So it's some other dependency or code change.

--- a/doc/source/tutorial/stats/plots/qmc_plot_conv_mc_sobol.py
+++ b/doc/source/tutorial/stats/plots/qmc_plot_conv_mc_sobol.py
@@ -49,7 +49,7 @@ sample_sobol_rmse = []
 rng = np.random.default_rng()
 
 def sampler_mc(x):
-    rng.random((x, case.dim))
+    return rng.random((x, case.dim))
 
 for ns in ns_gen:
     # Monte Carlo

